### PR TITLE
Adding code comments to remoteConfig.ts source

### PR DESCRIPTION
**Pull request by esmacik**
Adding source code comments to remoteConfig.js, mostly from the existing remoteConfig-externs.js file

### CODE REVIEW

File: src/providers/remoteConfig.ts [hunk 1]
```diff
@@ -107,27 +111,49 @@ export interface TemplateVersion {
...
+   * incremental, or a rollback operation. See
+   * [`RemoteConfigUpdateType`](/docs/reference/remote-config/rest/v1/Version#remoteconfigupdatetype)
+   * for valid values.
+   */
   updateType: string;
 
   /**
-   * The version number of the Remote Config template that was rolled back to,
-   * if the update was a rollback.
+   * The version number of the Remote Config template that was this update rolled
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]: This sounds awkward - maybe something like "If the update was a rollback, the version number of the Remote Config template it was rolled back to."?

File: src/providers/remoteConfig.ts [hunk 2]
```diff
@@ -107,27 +111,49 @@ export interface TemplateVersion {
...
+   * back to, if the update was a rollback.
    */
   rollbackSource?: number;
 }
 
+/**
+ * An interface representing metadata for a Remote Config account
+ * that performed an update. Contains the same fields as
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
 * that performed the update. Contains the same fields as
```

File: src/providers/remoteConfig.ts [hunk 3]
```diff
@@ -107,27 +111,49 @@ export interface TemplateVersion {
...
+ * [`RemoteConfigUser`](/docs/reference/remote-config/rest/v1/Version#remoteconfiguser).
+ */
 export interface RemoteConfigUser {
+  /** Name of the Remote Config account that performed an update. */
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
  /** Name of the Remote Config account that performed the update. */
```

File: src/providers/remoteConfig.ts [hunk 4]
```diff
@@ -107,27 +111,49 @@ export interface TemplateVersion {
...
   name?: string;
+
+  /** Email address of the Remote Config account that performed an update. */
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
  /** Email address of the Remote Config account that performed the update. */
```

File: src/providers/remoteConfig.ts [hunk 5]
```diff
@@ -107,27 +111,49 @@ export interface TemplateVersion {
...
   email: string;
+
+  /** Image URL of the Remote Config account that performed an update. */
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
  /** Image URL of the Remote Config account that performed the update. */
```

File: src/providers/remoteConfig.ts [hunk 6]
```diff
@@ -107,27 +111,49 @@ export interface TemplateVersion {
...
+   */
   updateUser: RemoteConfigUser;
 
-  /** A description associated with the particular Remote Config template. */
+  /** A description associated with this Remote Config template version. */
   description: string;
 
-  /** The origin of the caller. */
+  /**
+   * The origin of the caller, either the Firebase console or the Remote Config
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
   * The origin of the caller - either the Firebase console or the Remote Config
```

File: src/providers/remoteConfig.ts [hunk 7]
```diff
@@ -35,9 +35,13 @@ export const provider = 'google.firebase.remoteconfig';
...
- * Handle all updates (including rollbacks) that affect a Remote Config project.
- * @param handler A function that takes the updated Remote Config template
- * version metadata as an argument.
+ * Registers a function that triggers on Firebase Remote Config template
+ * update events.
+ *
+ * @param handler A function that takes the updated Remote Config
+ *   template version metadata as an argument.
+ *
+ * @return A Cloud Function which you can export and deploy.
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
 * @return A Cloud Function that you can export and deploy.
```

**Comment by thechenky** [comment]
@esmacik could you have more descriptive commit messages? Then it would be helpful to understand what changed in the commit just by looking at the name. Also, these changes are all so small, feel free to commit them all in one commit. I know we said small is good, but just letting you know it's okay to combine all these :)

**Comment by esmacik** [comment]
@thechenky definitely! And for those small commits I was just following the GitHub GUI. Sorry about that!

File: src/providers/remoteConfig.ts [hunk 8]
```diff
@@ -107,27 +111,50 @@ export interface TemplateVersion {
...
+   * for valid values.
+   */
   updateType: string;
 
   /**
-   * The version number of the Remote Config template that was rolled back to,
-   * if the update was a rollback.
+   *
+   * The version number of the Remote config template that this update rolled back to.
+   * Only applies of this update was a rollback.
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
   * Only applies if this update was a rollback.
```

File: src/providers/remoteConfig.ts [hunk 9]
```diff
@@ -107,27 +111,50 @@ export interface TemplateVersion {
...
+   * [`RemoteConfigUpdateType`](/docs/reference/remote-config/rest/v1/Version#remoteconfigupdatetype)
+   * for valid values.
+   */
   updateType: string;
 
   /**
-   * The version number of the Remote Config template that was rolled back to,
-   * if the update was a rollback.
+   *
+   * The version number of the Remote config template that this update rolled back to.
```
Comment (thechenky) [inline src/providers/remoteConfig.ts]:
```suggestion
   * The version number of the Remote Config template that this update rolled back to.
```

**Comment by thechenky** [comment]
Can you pull in master? Then I can merge it in.

**Pull request merged**